execution/executor-types/src/lib.rs

```diff
@@ -427,8 +427,11 @@ impl StateComputeResult {
     }
 
     pub fn transactions_to_commit_len(&self) -> usize {
-        // StateCheckpoint/BlockEpilogue is added if there is no reconfiguration
-        self.compute_status_for_input_txns().len()
+        self.compute_status_for_input_txns()
+            .iter()
+            .filter(|status| if let TransactionStatus::Keep(_) = status { true } else { false })
+            .count()
```

Comment:
hmm.. why has this changed?

Reply:
it was wrong, and it is used in tests. 

it was passing , because those tests had no block limit.

Reply:
hmmm.. wouldn't this make execution pipeline fail any time the block limit is hit??

Reply:
this function (transactions_to_commit_len) is only used in tests. 

transactions_to_commit is used throughout, and that one is correct.